content/exchange/artifacts/IRIS.Timeline.Add.yaml

```diff
@@ -171,13 +171,11 @@ sources:
                 else = if(condition=Color =~ "^orange",then="#FFAD4699",
                 else = "#31CE3699"))))))
 
-      LET eventDate = format(format="%v.%v",args=[parse_string_with_regex(string=Timestamp.String, regex='''(^.+)Z''').g1,
-                                          if(condition=Timestamp.Nanosecond, then=
-                                               parse_string_with_regex(
-                                               string=str(str=Timestamp.Nanosecond),
-                                               regex='''(^\d\d\d)''').g1
-                                               ,else="000")
-                                          ])
+      LET eventDate = format(format="%d-%02d-%02dT%02d:%02d:%02d.%03.f", args=[
+                                      Timestamp.Year, Timestamp.Month, Timestamp.Day,
+                                      Timestamp.Hour, Timestamp.Minute, Timestamp.Second,
+                                      Timestamp.Nanosecond / 1000000
```

Comment:
Are you sure this is correct? My understanding is that nanosecond is the total time in the minute in nanoseconds so it already includes the second part. Its not just the fractional part of the second.

Reply:
It seems to be only the fractional part

![Screenshot from 2024-07-06 13-08-35](https://github.com/Velocidex/velociraptor-docs/assets/9843422/007bd368-7798-496d-962c-2e0f1d193565)

https://pkg.go.dev/time#Time.Nanosecond